kimchi/src/circuits/lookup/constraints.rs

```diff
@@ -35,13 +32,14 @@ pub const ZK_ROWS: usize = 3;
 pub fn zk_patch<R: Rng + ?Sized, F: FftField>(
     mut e: Vec<F>,
     d: D<F>,
+    zk_rows: usize,
     rng: &mut R,
 ) -> Evaluations<F, D<F>> {
     let n = d.size();
     let k = e.len();
-    assert!(k <= n - ZK_ROWS);
-    e.extend((0..((n - ZK_ROWS) - k)).map(|_| F::zero()));
-    e.extend((0..ZK_ROWS).map(|_| F::rand(rng)));
+    assert!(k <= n - zk_rows);
+    e.extend((0..((n - zk_rows) - k)).map(|_| F::zero()));
```

Comment:
(very) minor: compute `n-zk_rows` just once

For later: is it worth thinking about not failing hard with assertions here (aka using an error monad) ?

Reply:
> is it worth thinking about not failing hard with assertions here

A failure here indicates an critical error on our side, so we really do want to fail hard instead of returning something to the user.